### DISCUSSION TITLE
Clover on Asus P9D-M

### DISCUSSION THREAD

**Discussion by jkaidor** [CloverHackyColor/CloverBootloader#340] [Q&A]
Hello,

   I'm trying to use Clover to boot my ancient ( 2013 ) workstation with an NVME SSD.    Windows 10 sees the SSD, but will not boot off it ( no driver in BIOS, duhh ).   I loaded Clover on a thumb drive and moved the NVME driver to the two appropriate places as documented on the Web.

   Here's what it's doing.  Restart the PC, get it to boot off the thumb drive, Clover starts up.  However, it will not boot *anything*.  I tried each of the list of "legacy drives" shown on the screen, and for each choice, I was presented a blank screen with an unmoving flashing cursor in the upper L/H corner.  

Reset button, try again... Same deal.  I tried for each and every drive presented on the screen.

If, however, I choose "Exit Clover", Windows then boots normally.

So - Clover does not boot the NVME drive, AND ALSO does not boot the existing OS drive.

So I suspect that the problem is not the NVME driver, but rather something more global.

 I have tried web searches, came up dry.    I used to be reasonably educated on boot/MBR issues, down to the level of bytes on the drive, but that was decades ago.  Have no clue as to what to even look at.

Clover seemed like a really good solution for my old but potent workstation ( 4 core XEON 3.2gHz, 32G of ECC RAM, absolutely silent build ).  Really don't want to build a new PC at the moment....

Where do I start looking?  Thanks in advance....

**Comment by serdeliuk** [comment]
You should try enabling debug boot log in config.plist then check what is doing, while enabled the boot process is slower than usual so be patient :) also in Clover menu you can press F2 to create a preboot log with infos on what was done until the menu appear.

**Reply by jkaidor** [reply]
Thanks,
I found each occurence of "debug" in config.plist and changed the value from "false" to "true".  Then booted.  It acted the same, and seemed just as fast.  A new directory appeared on the drive, with a single file "nolog" which was empty.

Had to pull the drive out of that PC and come upstairs to sit with my son in Zoom class.  Now that directory is gone.  Guess I should have ejected the drive first :).  Will try again this afternoon.

**Reply by serdeliuk** [reply]
Log files are located in `EFI/CLOVER/misc/` directory, press F2 too in clover menu just to be sure there is something :)

**Reply by jkaidor** [reply]
OK - just for yuks, I tried Clover on the thoroughly modern mini-PC I'm using in my son's classroom.  Clover works fine on this one.  It boots Windows from the internal NVME,  it creates the EFI/CLOVER/misc directory when I press F2.   I'll play around with it a bit on this PC just to see how it behaves when it works.

**Reply by jkaidor** [reply]
Here is the preboot.log from the PC that won't boot.

[preboot.log](https://github.com/CloverHackyColor/CloverBootloader/files/5818320/preboot.log)

**Reply by serdeliuk** [reply]
I think you don't have NTFS.efi driver, please find an NTFS.efi or NTFS-64.efi file and put it into drivers folder along the other efi files.

**Reply by serdeliuk** [reply]
The NVME hard disk it is saw by the clover, however, maybe does not contain a bootable partition, if you add the NTFS.efi you will see bootable partitions which are on NTFS too, assuming the windows ones.

**Reply by jkaidor** [reply]
I couldn't find an NTFS.efi in the  clover thumb drive.  Only "GrubNTFS.efi".   I did find another github project for an NTFS.efi.  Installed that, no difference.

**Comment by serdeliuk** [comment]
Try this one
[NTFS-64.zip](https://github.com/CloverHackyColor/CloverBootloader/files/5822724/NTFS-64.zip)

**Reply by debnath-d** [reply]
Thanks so much for providing the link to this NTFS driver! I was able to install Windows 11 on an old laptop without UEFI only due to this driver. The issue was that Clover just wouldn't detect my USB flash drive, which was very frustrating. This made it impossible to start the Windows installer from the USB flash drive. So, to install Windows I had to create a roughly 8 GB NTFS partition on the same SSD and copied the contents of the Windows installation iso to that partition. Now, the only way I could start the installer from the NTFS partition from Clover was with the help of this NTFS driver you linked.

I would like to know a few things: 

- How did you get this link (https://github.com/CloverHackyColor/CloverBootloader/files/5822724/NTFS-64.zip)? How do I search this repository to find this link myself (in the case that you didn't luckily provide this link in this discussion thread)?
- Why is this NTFS driver not part of the current releases? The NTFS driver provided in `EFI/CLOVER/drivers/off/GrubNTFS.efi` did not work for me. It didn't even detect the NTFS partition.

**Comment by jkaidor** [comment]
On 2021-01-15 12:15, marc wrote:
> Try this one
> NTFS-64.zip [1]
    Didn't work.  Behaves the same. :(.

**Reply by serdeliuk** [reply]
Well... i am not sure if i didn't missed something, but what exactly you intend to do? To boot an installed windows through Clover and you are unable to see the boot option? Maybe you need to unhide legacy menus in GUI > Custom > Scan by removing the # sign before Custom and Scan "in Clover 5128 default config"

**Comment by jkaidor** [comment]
On 2021-01-15 14:32, marc wrote:
> Well... i am not sure if i didn't missed something, but what exactly
> you intend to do?

*** Boot the system off a  Samsung NVME drive that I just obtained.  
Hoping to get a speed boost as compared to the existing SATA SSD.  The 
PC reads and writes it at run-time, but will not boot off it, because no
driver in the BIOS.

   I formatted the new drive as GPT partitions with gparted.  Then cloned 
each existing partition from the SATA boot drive with Macrium Reflect.

   Windows reads the new drive OK from the GUI.

   I wonder if the BIOS is just too old?  The existing BIOS is dated 
2013.
There are updates from ASUS up to 2018.  I haven't updated it yet, 
because their program to do the update requires 32-bit Windows.  I can
download an old Windows from Microsoft and load it onto a junk hard 
drive ( got lots of those since I converted everything to SSDs ) and 
apply
that update.  But that's a lot of work...

   OTOH, it might be time to update this PC to a newer MB.  I would need
a motherboard, a CPU, CPU cooler, DRAM...  Luckily, my copy of Win10 is
the retail kind, so I would be able to move it.

                        -






  To boot an installed windows through Clover and you

**Comment by serdeliuk** [comment]
@jkaidor please press again F2 and provide the log file with NTFS driver installed, also provide the config.plist.

Your Clover clearly see the NVME drive, find 3 partitions on the same drive

2:428  0:000  - [08]: Volume: PciRoot(0x0)\Pci(0x1C,0x4)\Pci(0x0,0x0)\NVMe(0x1,B2-30-90-01-5B-38-25-00)
2:428  0:000  - [09]: Volume: PciRoot(0x0)\Pci(0x1C,0x4)\Pci(0x0,0x0)\**NVMe**(0x1,B2-30-90-01-5B-38-25-00)\HD(1,GPT,A6B136F0-7565-4806-AA55-CEEA19C2CCCA,0x800,0x32000)
2:428  0:000  - [10]: Volume: PciRoot(0x0)\Pci(0x1C,0x4)\Pci(0x0,0x0)\**NVMe**(0x1,B2-30-90-01-5B-38-25-00)\HD(2,GPT,22F4D42E-4E71-4923-A578-3240CE211FE9,0x32800,0x73B21000)
2:428  0:000  - [11]: Volume: PciRoot(0x0)\Pci(0x1C,0x4)\Pci(0x0,0x0)\**NVMe**(0x1,B2-30-90-01-5B-38-25-00)\HD(3,GPT,B6B11A74-3B49-4394-90DB-A7222BAAE051,0xE7608800,0x1800000)


But does not seems to find any bootable partition and ends with only one bootable entry from Sata

2:446  0:001  === [ GetEfiBootDeviceFromNvram ] ===============
2:446  0:000   - efi-boot-device-data: PciRoot(0x0)\Pci(0x1F,0x2)\**Sata**(0x0,0xFFFF,0x0)\HD(1,MBR,0x15777835,0x800,0x32000)
2:446  0:000    - Volume: 'PciRoot(0x0)\Pci(0x1F,0x2)\Sata(0x0,0xFFFF,0x0)\HD(1,MBR,0x15777835,0x800,0x32000)'
2:446  0:000    - LoaderPath: ''
2:446  0:000  === [ FindStartupDiskVolume ] ===================
2:446  0:000    - Volume: partition = PciRoot(0x0)\Pci(0x1F,0x2)\**Sata**(0x0,0xFFFF,0x0)\HD(1,MBR,0x15777835,0x800,0x32000)
2:446  0:000     - searching for that partition
2:446  0:000      - found entry 0. 'Boot  from Legacy HD1', Volume 'Legacy HD1'

**Comment by serdeliuk** [comment]
I have observed that your SATA HDD it is partitioned as MBR but your NVME as GPT so probably this is the issue, you should clone entire SATA-HDD over the NVME including MBR not just partitions, because your SATA use MBR you are missing exactly the BOOT flags on your newly GPT.
Try to clone entire SATA over NVME
